test/Performance/lib/PDOSqlsrvUtil.php

```diff
@@ -0,0 +1,228 @@
+        self::bindParam( $stmt, 1, $values[0], PDO::PARAM_STR);
+        self::bindParam( $stmt, 2, $values[1], PDO::PARAM_STR);
+        self::bindParam( $stmt, 3, $values[2], PDO::PARAM_INT);
+        self::bindParam( $stmt, 4, $values[3], PDO::PARAM_STR);
+        self::bindParam( $stmt, 5, $values[4], PDO::PARAM_STR);
+        self::bindParam( $stmt, 6, $values[5], PDO::PARAM_STR);
+        self::bindParam( $stmt, 7, $values[6], PDO::PARAM_STR);
+        self::bindParam( $stmt, 8, $values[7], PDO::PARAM_LOB);
+        self::bindParam( $stmt, 9, $values[8], PDO::PARAM_LOB);
+        self::bindParam( $stmt, 10, $values[9], PDO::PARAM_STR);
+    }
+
+    public static function createCRUDTable( $conn, $tableName ){
+        $fields = array(
+                    "vstring" => "VARCHAR(64)",
+                    "nvstring" => "NVARCHAR(64)",
+                    "num" => "int",
+                    "dttwo" => "DATETIME2",
+                    "string" => "CHAR(64)",
+                    "nstring" => "NCHAR(64)",
+                    "real" => "NUMERIC",
+                    "bin" => "BINARY(16)",
+                    "vbin" => "VARBINARY",
+                    "dtoffset" => "DATETIMEOFFSET");
+        $params = "";
+        foreach( $fields as $fieldName => $type ){
+            $params .= $fieldName." ".$type.",";
+        }
+        $params = rtrim($params,", ");
+        self::createTable( $conn, $tableName, $params );
+    }
+
+    private function createDatabase( $conn, $dbName ){
+        $sql = "CREATE DATABASE $dbName";
+        $conn->exec( $sql );
+    }
+
+    public static function dropDatabase( $conn, $dbName ){
+        $sql = "USE MASTER;DROP DATABASE $dbName";
+        $conn->exec( $sql );
+    }
+
+    public static function createTable( $conn, $tableName, $params ){
+        $sql = "CREATE TABLE $tableName ($params)";
+        $conn->exec( $sql );
+    }
+
+    public static function dropTable( $conn, $tableName ){
+        $sql = "DROP TABLE $tableName";
+        $conn->exec( $sql );
+    }
+
+    private function useDatabase( $conn, $dbName ){
+        $sql = "USE $dbName";
+        $conn->exec( $sql );
+    }
+
+    private function createStoredProc( $conn, $procName, $params, $text ){
+        $sql = "CREATE PROCEDURE $procName $params AS $text";
+        $conn->exec( $sql );
+    }
+
+    private function dropStoredProc( $conn, $procName ){
+        $sql = "DROP PROCEDURE $procName";
+        $conn->exec( $sql );
+    }
+
+    private function query( $conn, $sql ){
+        try{
+            return $conn->query( $sql );
+        }
+        catch( PDOException $e ){
+            var_dump( $e );
+            exit;
+        }
+    }
+
+    private function fetch( $stmt ){
+        return $stmt->fetch();
+    }
+
+    private function prepare( $conn, $sql ){
+        try{
+            $stmt = $conn->prepare( $sql );
+            if( $stmt === false ){
+                die( "Failed to prepare\n");
+            }
```

Comment:
this `($stmt === false)` block is not needed. At connection, you did `$conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);`, so if prepare fails, it returns a PDOException object